### PR TITLE
[gh-265] Add the IEx addon

### PR DESCRIPTION
**Pull request by byhbt**
Closes https://github.com/nimblehq/elixir-templates/issues/265

## What happened 👀

- [x] Add the `IEx` addon to generate the default `.iex.exs` add the project root directory
- [x] Add `elixir_ls` to the generated `.gitignore`
- [x] Remove the `RFC template` in the Issue section. From now on, to propose a new RFC, we will use [GitHub Discussion](https://github.com/nimblehq/elixir-templates/discussions) instead.

## Insight 📝

The `WebApp.Repo` is only loaded for the web project. 

## Proof Of Work 📹

image

image

The tests should pass.

### CODE REVIEW

**Review by andyduong1920 — APPROVED** [review]

**Review by liamstevens111 — COMMENTED** [review]
Few questions otherwise I think it's a good addition to the template 👍

**Review by Nihisil — APPROVED** [review]

**Review by nvminhtue — APPROVED** [review]

**Review by hanam1ni — APPROVED** [review]

File: priv/templates/nimble_template/.iex.exs.eex [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Load personal IEx config
+import_file_if_available("~/.iex.exs")
+<%= if web_project? do %>
+import_if_available(Ecto.Query)
+
+alias <%= base_module %>.Repo
+<% end %>
+IEx.configure(
+  history_size: 1000,
```
[the remark this answers is not in the document]
Comment (byhbt) [reply priv/templates/nimble_template/.iex.exs.eex:9]: Here is the default of `history_size` https://github.com/elixir-lang/elixir/blob/main/lib/iex/mix.exs#L19

File: priv/templates/nimble_template/.iex.exs.eex [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    pretty: true,
+    limit: :infinity,
+    charlists: :as_lists
+  ],
+  colors: [
+    syntax_colors: [
+      number: :light_yellow,
+      atom: :light_cyan,
+      string: :light_black,
+      boolean: :red,
```
[the remark this answers is not in the document]
Comment (byhbt) [reply priv/templates/nimble_template/.iex.exs.eex:20]: Here is the default output:

image

Compare with the customized output:

image

As you can see that the colors have improved for readability. So as a developer, you can get started with that and don't take more time in finding colors to customization.

[the remark this answers is not in the document]
Comment (liamstevens111) [reply priv/templates/nimble_template/.iex.exs.eex]: @byhbt So if we confirmed the colours in your terminal (all the same) are not the default colours and they actually have distinct colours as previously mentioned (although bools and nil are same 😢 ) then yeah I think should be left as default IMO. 👍

File: priv/templates/nimble_template/.iex.exs.eex (hunk 1 above)
[the remark this answers is not in the document]
Comment (liamstevens111) [reply priv/templates/nimble_template/.iex.exs.eex]: I see, you need to have history enabled first right, ie

`iex --erl “-kernel shell_history enabled”`
https://hexdocs.pm/iex/1.14/IEx.html#module-shell-history

**Review by liamstevens111 — APPROVED** [review]
Just small comment about history otherwise good to go

File: priv/templates/nimble_template/.iex.exs.eex (hunk 1 above)
[the remark this answers is not in the document]
Comment (byhbt) [reply priv/templates/nimble_template/.iex.exs.eex]: yes, we need to enable it on the system as a whole.

```
export ERL_AFLAGS="-kernel shell_history enabled"
```

**Pull request merged**